alibi_detect/saving/schemas.py

```diff
@@ -73,10 +72,15 @@ class Config:
         extra = 'allow'  # Allow extra fields
 
 
-class MetaData(CustomBaseModel):
-    version: str = __version__
-    config_spec: str = __config_spec__
-    version_warning: Optional[bool] = None  # None instead of False as same effect in logicals but hides from config
+class MetaData(CustomBaseModelWithKwargs):
+    name: str
+    online: bool
+    data_type: Optional[str]
+    detector_type: str
```

Comment:
should we add the `Literal` type here?

Reply:
@jklaise I actually just realised, I agree we should update to `Literal`. Only issue is the docstrings for `data_type` are out of date throughout the library i.e. we say `Optionally specify the data type (tabular, image or time-series)`. 

We should probably agree on all the permitted options and update the docstrings/docs at the same time. I can do as part of this PR if you think its worth it?

Options I can think of are `tabular`, `image`, `time-series`, `text`, `graph`. `graph` isn't included in our [algo overview](https://docs.seldon.io/projects/alibi-detect/en/stable/overview/algorithms.html) so don't know if we should include...

Reply:
Would prefer a different PR so as to keep things separate. I'm not sure how the docstrings currently relate to the functionality of passing `data_type`, i.e. what does a user specify exactly if we haven't even defined what they can do?

Reply:
We should also review the need and use case behind the `data_type` kwarg. I've included this in https://github.com/SeldonIO/alibi-detect/issues/567.